### ISSUE TITLE
Add support for transit-java-0.8.327 and its defaultWriteHandler

### ISSUE THREAD

**Issue by mikub** [cognitect/transit-clj#37]
As this is quite important functionality I guess transit-clj should not be left behind. 

This should not be hard to implement, happy to do a pull request (but I think you don't accept those from non-contrib ppl, right?).

Changes required would basically be something like:
```clojure
(defn writer
  "Creates a writer over the provided destination `out` using
   the specified format, one of: :msgpack, :json or :json-verbose.
   An optional opts map may be passed. Supported options are:
   :handlers - a map of types to WriteHandler instances, they are merged
   with the default-handlers and then with the default handlers
   provided by transit-java
   :unsupported-handler - a default WriteHandler that will be used for types that don't have a handler 
   (if none is  used an Exception is thrown by default). This way types can be handled (e.g. 
   converted to string etc.) so as they can still be part of the payload. A caution should be taken when 
   leveraging this feature as these types will likely not be fully unmarshalled to their original type."
  ([out type] (writer out type {} nil))
  ([^OutputStream out type {:keys [handlers unsupported-handler]}]
     (if (#{:json :json-verbose :msgpack} type)
       (let [handler-map (if (instance? HandlerMapContainer handlers)
                           (handler-map handlers)
                           (merge default-write-handlers handlers))]
         (Writer. (TransitFactory/writer (transit-format type) out handler-map unsupported-handler)))
(throw (ex-info "Type must be :json, :json-verbose or :msgpack" {:type type})))))
```
As outlined in the transit-java readme, a default write handler can look and be used like this:
```java
WriteHandler customDefaultWriteHandler = new WriteHandler() {
    @Override
    public String tag(Object o) { return "unknown"; }
    @Override
    public Object rep(Object o) { return o.toString(); }
    @Override
    public String stringRep(Object o) { return o.toString(); }
    @Override
    public WriteHandler getVerboseHandler() { return this; }
};
OutputStream out = new ByteArrayOutputStream();
Writer w = TransitFactory.writer(TransitFactory.Format.JSON, out, customDefaultWriteHandler);
w.write(new Point(37,42));
System.out.print(out.toString());
// => "[\"~#unknown\",\"Point at 37, 42\"]"
```

Imho this is a super useful feature and a vital use case without which transit _cannot_ be used in enterprise environment - where one simply cannot fully control what data types are used by the other (upstream/downstream) applications. It has to be handled more gracefully then Runtime error / 500.

**Comment by puredanger** [comment]
This is a reasonable suggestion - any reason for `unsupported-handler` rather than `default-handler`?  Seems like we should stick with the same word here.

**Comment by puredanger** [comment]
Released in transit-clj 0.8.319
